packages/core/src/analytics/redux/middlewares/bag.js

```diff
@@ -278,34 +277,31 @@ export default (analyticsInstance, customActionTypes) => {
           oldQuantity: previousBagItem?.quantity,
           ...getBagData(action),
         };
-        let eventType = null;
-
-        // Track ga4 update event
-        // Data is being cloned here because it will be mutated later
-        analyticsInstance.track(eventTypes.PRODUCT_UPDATED, { ...data });
-
-        if (data.oldSize && data.oldSize !== data.size) {
-          const removedProductData = { ...data };
-          removedProductData.size = data.oldSize;
-          removedProductData.quantity = data.oldQuantity;
-
-          analyticsInstance.track(
-            eventTypes.PRODUCT_REMOVED_FROM_CART,
-            removedProductData,
-          );
 
-          data.oldQuantity = 0;
+        // product updated will only triggered if provenience is from BAG
+        if (data.from === fromParameterTypes.BAG) {
+          // Track ga4 update event
+          // Data is being cloned here because it will be mutated later
+          analyticsInstance.track(eventTypes.PRODUCT_UPDATED, { ...data });
         }
 
+        let eventType = null;
         // Check if the quantity difference is less than it was in bag - use PRODUCT_REMOVED_FROM_CART in that case
         if (data.oldQuantity && data.quantity < data.oldQuantity) {
           eventType = eventTypes.PRODUCT_REMOVED_FROM_CART;
-        } else {
+        } else if (data.oldQuantity && data.quantity > data.oldQuantity) {
           eventType = eventTypes.PRODUCT_ADDED_TO_CART;
         }
 
         data.quantity = Math.abs(data.quantity - (data.oldQuantity || 0));
-        analyticsInstance.track(eventType, { ...data, oldQuantity: undefined });
+
+        // If eventType was set on previous ifs, then track
+        if (eventType !== null) {
+          analyticsInstance.track(eventType, {
+            ...data,
+            oldQuantity: undefined,
```

Comment:
Should we do this here? 
With this, we're not passing the value to integrations that might have interest in this property. If GA4 does not need `oldQuantity` in these events, then it should be GA4's responsibility to ignore the property, instead of forcing an `undefined` on the analytics track layer.

Reply:
On eventType Product added/removed from cart (and wishlist too) oldQuantity isn't required for these events, so it could be sent any value to track. But in this case, I think we should send undefined because for me, when you put the new quantity value with previous calculation, that result on quantity added or removed of some product (not real quantity), then use the real oldQuantity value seems wrong information.
EG: when you have an product on bag with quantity 10, then you remove 2, will be triggered PRODUCT_REMOVED_FROM_CART, with quantity 2, instead of 8. If we introduce oldQuantity value on analytics, it could induce that you have 10 products, then you remove products and now the quantity it's 2. Beside that we have other events with similar payloads, that quantity and oldQuantity represents different things, such quantity here it's the changedQuantity, but in other events, quantity represents bag or wishlist product quantity.
Other relevant information it's that we already did this(send undefined) on previous code, the only thing that changed it's the if around.

What you think of maintain this here in main the undefined, and change in next to append one field to payload?
- changedQuantity: moved quantity (actual quantity on this events)
- quantity: remove calculation, then use the real value of quantity.
- oldQuantity: send old quantity values.

And in each integration, on this method, map from changedQuantity instead of map from quantity.
With this payload, I think would be more compreensive, because it could be more clear, because in every event, we could know that quantity always represents same thing on each event, instead of what we have now (in some events, quantity represents quantity of items, then in other events, it represents quantity of changed items).